shaft_element.py

```diff
@@ -1,478 +1,461 @@
+        Inner diameter of the element.
+    o_d : float
+        Outer diameter of the element.
+    material : ross.material
+        Shaft material.
+    n : int, optional
+        Element number (coincident with it's first node).
+        If not given, it will be set when the rotor is assembled
+        according to the element's position in the list supplied to
+        the rotor constructor.
+    axial_force : float
+        Axial force.
+    torque : float
+        Torque.
+    shear_effects : bool
+        Determine if shear effects are taken into account.
+        Default is False.
+    rotary_inertia : bool
+        Determine if rotary_inertia effects are taken into account.
+        Default is False.
+    gyroscopic : bool
+        Determine if gyroscopic effects are taken into account.
+        Default is False.
+    Returns
+    -------
+    Attributes
+    ----------
+    Poisson : float
+        Poisson coefficient for the element.
+    A : float
+        Element section area.
+    Ie : float
+        Ie is the second moment of area of the cross section about
+        the neutral plane Ie = pi*r**2/4
+    phi : float
+        Constant that is used according to [1]_ to consider rotary
+        inertia and shear effects. If these are not considered phi=0.
+    References
+    ----------
+    .. [1] 'Dynamics of Rotating Machinery' by MI Friswell, JET Penny, SD Garvey
+       & AW Lees, published by Cambridge University Press, 2010 pp. 158-166.
+    Examples
+    --------
+    >>> from ross.materials import steel
+    >>> le = 0.25
+    >>> i_d = 0
+    >>> o_d = 0.05
+    >>> Euler_Bernoulli_Element = ShaftElement(le, i_d, o_d, steel,
+    ...                                        shear_effects=False, rotary_inertia=False)
+    >>> Euler_Bernoulli_Element.phi
+    0
+    >>> Timoshenko_Element = ShaftElement(le, i_d, o_d, steel,
+    ...                                   rotary_inertia=True,
+    ...                                   shear_effects=True)
+    >>> Timoshenko_Element.phi
+    0.08795566502463055
+    """
+
+    def __init__(
+        self,
+        L,
+        i_d,
+        o_d,
+        material,
+        n=None,
+        axial_force=0,
+        torque=0,
+        shear_effects=True,
+        rotary_inertia=True,
+        gyroscopic=True,
+        shear_method_calc=None,
+    ):
+
+        self.shear_effects = shear_effects
+        self.rotary_inertia = rotary_inertia
+        self.gyroscopic = gyroscopic
+
+        self._n = n
+        self.n_l = n
+        self.n_r = None
+        if n is not None:
+            self.n_r = n + 1
+
+        self.shear_method_calc = shear_method_calc
+
+        self.L = float(L)
+
+        # diameters
+        self.i_d = float(i_d)
+        self.o_d = float(o_d)
+        self.i_d_l = float(i_d)
+        self.o_d_l = float(o_d)
+        self.i_d_r = float(i_d)
+        self.o_d_r = float(o_d)
+
+        self.material = material
+        self.material_name = material.name

```

Comment:
Remove these attributes:

```python
self.material_name = material.name
self.E = material.E
self.G_s = material.G_s
self.Poisson = material.Poisson
self.color = material.color
self.rho = material.rho
```
since we are going to access them through the material object.

Reply:
These changes are shown in this commit [here](https://github.com/rodrigomoliveira1/ross/commit/100f3bb8f6ba6076a7c77500c59c8cbb82ae218a).